tests/util/caches/test_deferred_cache.py

```diff
@@ -53,7 +53,9 @@ def check1(r):
             self.assertTrue(set_d.called)
             return r
 
-        get_d.addCallback(check1)
+        # TODO: Actually ObservableDeferred *doesn't* run its tests in order on py3.8.
+        #   maybe we should fix that?
+        # get_d.addCallback(check1)
```

Comment:
Yeah this is confusing. I'm not sure it is worth fixing (in this PR, at least). Do you see any downsides to this in practice?

Reply:
well, at some level I suspect it means we end up processing requests that arrived later before those that arrived earlier, which isn't exactly forbidden but does feel like it might lead to some surprising behaviour.

But likewise, I'm not going to rush into fixing it right now.